src/v2/client/startLoginFlow.js

```diff
@@ -22,24 +23,41 @@ const handleProxyIdxResponse = async (settings) => {
   });
 };
 
-export async function startLoginFlow(settings, appState) {
+export async function startLoginFlow(settings) {
   // Return a preset response
   if (settings.get('proxyIdxResponse')) {
     return handleProxyIdxResponse(settings);
   }
 
-  // Use stateToken
+  // Use stateToken from session storage if exists
+  // See more details at ./docs/use-session-token-prior-to-settings.png
+  const stateHandleFromSession = sessionStorageHelper.getStateHandle();
+  if (stateHandleFromSession) {
+    return introspect(settings, stateHandleFromSession)
+      .then(() => {
+        // 1. abundant the settings.stateHandle given session.stateHandle is still valid
```

Comment:
nit: Do these numbers point to something specific? If not, we can probably remove the numbers here.

Reply:
not really but my personal preference to dictates the logics/steps of the implementations.